identity-service/src/notifications/trendingTrackProcessing.js

```diff
@@ -0,0 +1,172 @@
+const axios = require('axios')
+const moment = require('moment')
+
+const models = require('../models')
+const { logger } = require('../logging')
+const config = require('../config.js')
+const {
+  deviceType,
+  notificationTypes
+} = require('./constants')
+const { publish } = require('./notificationQueue')
+const {
+  decodeHashId,
+  shouldNotifyUser
+} = require('./utils')
+const { fetchNotificationMetadata } = require('./fetchNotificationMetadata')
+const {
+  notificationResponseMap,
+  notificationResponseTitleMap,
+  pushNotificationMessagesMap
+} = require('./formatNotificationMetadata')
+
+const notifDiscProv = config.get('notificationDiscoveryProvider')
+
+const TRENDING_TIME = Object.freeze({
+  DAY: 'day',
+  WEEK: 'week',
+  MONTH: 'month',
+  YEAR: 'YEAR'
+})
+
+const TRENDING_GENRE = Object.freeze({
+  ALL: 'all'
+})
+
+const getTimeGenreActionType = (time, genre) => `${time}:${genre}`
+
+// The minimum time in hrs between notifications
+const TRENDING_INTERVAL = 3
+
+// The highest rank for which a notification will be sent
+const MAX_TOP_TRACK_RANK = 10
+
+async function getTrendingTracks () {
+  try {
+    // The owner info is then used to target listenCount milestone notifications
+    let params = new URLSearchParams()
+    params.append('time', TRENDING_TIME.WEEK)
+    params.append('limit', MAX_TOP_TRACK_RANK)
+
+    const trendingTracksResponse = await axios({
+      method: 'get',
+      url: `${notifDiscProv}/v1/full/tracks/trending?time=week`,
+      params,
+      timeout: 10000
+    })
+    const trendingTracks = trendingTracksResponse.data.data.map((track, idx) => ({
+      trackId: decodeHashId(track.id),
+      rank: idx + 1,
+      userId: decodeHashId(track.user.id)
+    }))
+    const blocknumber = trendingTracksResponse.data.latest_indexed_block
+    return { trendingTracks, blocknumber }
+  } catch (err) {
+    console.log(err)
+    return null
+  }
+}
+
+async function processTrendingTracks (audiusLibs, blocknumber, trendingTracks, tx) {
```

Comment:
nit: since this function does the majority of the work, can we leave a high level comment explaining the sequence of steps it takes, or add a comment linking to the corresponding readme if it's explained there?